### DISCUSSION TITLE
Preferred way store additional context about the error

### DISCUSSION THREAD

**Discussion by Glyphack** [RealOrangeOne/django-tasks#125] [Q&A]
I'm using django-tasks to run a task that sometimes fails. When it fails I have access to stacktrace and the error raised. But if the exception itself has more information it's not captured, this is behavior is documented.

But I would like a way to capture additional information about the task failure.

In my code I raise the following exception:

```
raise subprocess.CalledProcessError(
	process.returncode, cmd, output=stdout, stderr=stderr
)
```

And only the exception type and stacktrace are stored in the database.

I'm wondering what would be a good way to store the additional information from the error in the task.
This could be solved by logs as well. For example prefixing all the logs with `task_id` and then search the logs.

**Comment by RealOrangeOne** [comment]
I don't think there's a "preferred" way. There are a few different options. Presumably you need to change application behaviour rather than this just being useful for debugging?

Smuggling the data out and saving it to a DB model might be a better way, or possibly changing the return type so the task always succeeds but then stores extra context.

Alternatively, handle `CalledProcessError`, identify the cases you'd need to handle inside the task, and raise specific exceptions instead.
